wasmedge-ggml-llama-interactive/README.md

```diff
@@ -134,21 +134,35 @@ The total cost of 3 apples would be 15 dollars. Each apple costs 5 dollars, so 3
 
 ## Parameters
 
-Currently, we support the following parameters:
+Currently, we support setting llama options using `set_input` with index 1.
+You can pass the JSON string as a `Vec<u8>` type to `set_input`.
 
-- `LLAMA_LOG`: Set it to a non-empty value to enable logging.
-- `LLAMA_N_CTX`: Set the context size, the same as the `--ctx-size` parameter in llama.cpp (default: 512).
-- `LLAMA_N_PREDICT`: Set the number of tokens to predict, the same as the `--n-predict` parameter in llama.cpp (default: 512).
+Supported parameters include:
 
-These parameters can be set by adding the following environment variables before the `wasmedge` command:
+- `enable-log`: Set it to true to enable logging. (default: `false`)
+- `stream-stdout`: Set it to true to print the inferred tokens to standard output. (default: `false`)
+- `ctx-size`: Set the context size, the same as the `--ctx-size` parameter in llama.cpp. (default: `512`)
+- `n-predict`: Set the number of tokens to predict, the same as the `--n-predict` parameter in llama.cpp. (default: `512`)
+- `n-gpu-layers`: Set the number of layers to store in VRAM, the same as the `--n-gpu-layers` parameter in llama.cpp. (default: `0`)
+- `reverse-prompt`: Set it to the token at which you want to halt the generation. Similar to the `--reverse-prompt` parameter in llama.cpp.  (default: `""`)
+- `batch-size`: Set the number of batch size for prompt processing, the same as the `--batch-size` parameter in llama.cpp.  (default: `512`)
+
+(For more detailed usage instructions regarding the parameters, please refer to [WasmEdge](https://github.com/WasmEdge/WasmEdge/blob/master/plugins/wasi_nn/ggml.cpp).)
+
+For convenience, these parameters could be set by adding the environmental variables in this example.
+The environmental variables are handled by Rust. (Due to the limitation of environmental variables, beware of the `-` and `_` in the variable name.)
 
 ```bash
-LLAMA_LOG=1 LLAMA_N_CTX=1024 LLAMA_N_PREDICT=128 \
 wasmedge --dir .:. \
-  --nn-preload default:GGML:CPU:llama-2-7b-chat.Q5_K_M.gguf \
+  --env stream_stdout=false \
+  --env enable_log=false \
+  --env ctx_size=512 \
+  --env n_predict=512 \
+  --env n_gpu_layers=0 \
+  --nn-preload default:GGML:AUTO:llama-2-7b-chat.Q5_K_M.gguf \
   wasmedge-ggml-llama-interactive.wasm default
 ```
 
 ## Credit
 
-The WASI-NN ggml plugin embedded [`llama.cpp`](git://github.com/ggerganov/llama.cpp.git@b1217) as its backend.
+The WASI-NN ggml plugin embedded [`llama.cpp`](git://github.com/ggerganov/llama.cpp.git@b1309) as its backend.
```

Comment:
b1383

Reply:
Update the README and add a link to [WasmEdge/LICENSE.spdx](https://github.com/WasmEdge/WasmEdge/blob/master/LICENSE.spdx) to avoid frequent updates here.